# 데이터분석방법론2 - 제5강 과제

### 학    번 : 201685-010100
###  성    명 : 지 용 기
###  연 락 처 : 010-9828-0332, braveji@hanmail.net
---

![](chap05_01.jpg)

### a.로지스틱 회귀모형을 이용하여 열 문제가 발생할 확률에 대한 기온의 효과를 적합하고 적합 결과를 해석하라.

In [6]:
temp <- c( 66, 70, 69, 68, 67, 72, 73, 70,
          57, 63, 70, 78, 67, 53, 67, 75,
          70, 81, 76, 79, 75, 76, 58 )
td <- factor( c( 0, 1, 0, 0, 0, 0, 0, 0,
                 1, 1, 1, 0, 0, 1, 0, 0,
                 0, 0, 0, 0, 1, 0, 1 ), levels=c(0, 1) )

space_accident <- data.frame(temp, td )
head( space_accident )
summary( space_accident )

temp td
1   66  0
2   70  1
3   69  0
4   68  0
5   67  0
6   72  0

      temp       td    
 Min.   :53.00   0:16  
 1st Qu.:67.00   1: 7  
 Median :70.00         
 Mean   :69.57         
 3rd Qu.:75.00         
 Max.   :81.00         

In [7]:
space_accident.glm <- glm( td ~ temp  , binomial(link="logit"),   data=space_accident )
summary( space_accident.glm )


Call:
glm(formula = td ~ temp, family = binomial(link = "logit"), data = space_accident)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.0611  -0.7613  -0.3783   0.4524   2.2175  

Coefficients:
            Estimate Std. Error z value Pr(>|z|)  
(Intercept)  15.0429     7.3786   2.039   0.0415 *
temp         -0.2322     0.1082  -2.145   0.0320 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 28.267  on 22  degrees of freedom
Residual deviance: 20.315  on 21  degrees of freedom
AIC: 24.315

Number of Fisher Scoring iterations: 5


### 해석

- ** \\( logit[ \pi(x) ]  = 15.0429  -0.2322x \\)  라는 모형으로 적합되어 있으며, 
\\( \alpha 와 \beta \\)  의 유의수준이 0.05보다 작으므로 모형은 유의미함.  ** 
- ** e ^ ( 0.232 ) = 1.26137197799817 이므로 온도가 한 단위 떨어지면, 1.26만큼 사고 확율이 증가함. **

In [8]:
 exp( 0.2322 )

[1] 1.261372

### b. 챌린저호 비행시의 기온이 31℉에서 열문제가 발생할 확률을 예측하라.

- **  \\( logit[ \pi( 31 ) ]  = 15.0429 - 0.2322 * 31 \\)  **
- **  \\( log[ \pi( 31 ) / ( 1 - \pi( 31 ) ) ]  = 15.0429 - 0.2322 * 31 \\)  **
- **  \\(  \pi( 31 ) / ( 1 - \pi( 31 ) )   =  e ^ ( 15.0429 - 0.2322 * 31 )  \\)  **

- **  \\(  \pi( 31 ) =  e ^ ( 15.0429 - 0.2322 * 31 )  \\)  **
    - **  \\(      /         ( 1 +  e ^ ( 15.0429 - 0.2322 * 31 ) ) \\)  **
- ** 이므로 0.9996 %  **

In [9]:
exp( 15.0429 - 0.2322 * 31  ) / ( 1 + exp( 15.0429 - 0.2322 * 31  ) ) 

[1] 0.9996083

### c. 기온이 몇 도일 때에 예측확률이 0.50이 되는가? 구한 기온값에서 기온이 1도 증가함에 따라 나타나는 예측확률의 변화를 나타내는 선형근사식을 구하라.

- ** \\(  x =  - \alpha / \beta  \\) ** 이므로
- ** \\(  -15.0429  / -0.2322  =    \\) 64.7842377260982   ** 

In [10]:
-15.0429  / -0.2322

[1] 64.78424

- ** 예측확률 0.5 이후에 기온 1도 증가함에 따라서 1.26 만큼 사고 확율이 증가하므로 **
- ** \\(  \pi( x ) =  0.5 + 0.0126 x  \\)  **

### d. 열문제가 발생할 오즈에 대한 기온의 효과를 해석하라

- ** x + 1 에서의 오즈는 X에서의 오즈의 \\(  e^\beta \\) 를 곱한것과 같으므로 **
- ** 기온 한단위가 증가할때마다  \\(  e^( -0.2322 ) \\) ** = 0.79278 만큼 오즈가 증가함.

In [11]:
exp( -0.2322 ) 

[1] 0.7927875

### e. (i) 왈드 검정과 (ii) 가능도비 검정을 이용하여 기온의 효과에 대한 가설검정을 하라

- 왈드 검정 신뢰 구간 
- 0.95% 신뢰구간에 0을 포함하지 않으므로  \\( \beta  \\) 값은 유효함.

In [12]:
confint.default(  space_accident.glm, level = 0.95 )

2.5 %      97.5 %
(Intercept)  0.5810523 29.50475096
temp        -0.4443022 -0.02002324

- 가능도비 검정 신뢰 구간 
- 0.95% 신뢰구간에 0을 포함하지 않으므로  \\( \beta  \\) 값은 유효함.

In [13]:
confint(  space_accident.glm, level = 0.95 )

Waiting for profiling to be done...


2.5 %      97.5 %
(Intercept)  3.3305848 34.34215133
temp        -0.5154718 -0.06082076